scrabble.js

```diff
@@ -1,16 +1,136 @@
 
+
+
 const Scrabble = {
   score(word) {
+    if(word === "" || word == null) {
+      throw new Error ("Must enter a word with at least one character")
+    }
+
+    let wordArray = word.toUpperCase().split(``);
+    let wordTotal = 0
+
+    if(wordArray.length > 7) {
+      throw new Error ("Word is too long for Scrabble, max 7 letters");
+    }
+    if(!/^[a-zA-Z]+$/.test(word)) {
+      throw new Error ("Invalid character in the word");
+    }
 
+
+    wordArray.forEach(function(letter){
+      if ([`A`, `E`, `I`, `O`, `U`, `L`, `N`, `R`, `S`, `T`].includes(letter)) {
+        wordTotal += 1;
+      }
```

Comment:
I'm not a big fan of organizing the letter scores in this way. It's not very DRY (you've had to type `if ([...].includes(letter)` many times), but more importantly the data is intertwined with the code that uses it, which means there's no easy way to reuse this information.

For example, what if the player had a hand full of tiles, and you wanted to be able to display the value of each of those letters? You would have to copy/paste this switch in that code, or at least extract it to a method.

A better approach would be to store the scores in a hash:

```javascript
const LETTER_VALUES = {
  a: 1,
  b: 3,
  // ...
}
```

then here you could use it like this:

```javascript
word.forEach((letter) => {
  const value = LETTER_VALUES[letter.toLowerCase()];
  if (value) {
    totalScore += letter.toLowerCase();
  } else {
    throw new Error(`Invalid character ${letter}`);
  }
}
```